Remotion/ObjectBinding/Web.ClientScript/Scripts/BocList.ts

```diff
@@ -0,0 +1,779 @@
+  //  bocList: The BocList whose rows should be unselected.
+  //  isRowHighlightingEnabled: true to update the row's css-class.
+  public static UnselectAllRows(bocList: HTMLElement, isRowHighlightingEnabled: boolean): void
+  {
+    var selectedRows = BocList._selectedRows[bocList.id];
+    for (var rowID in selectedRows?.Rows)
+    {
+      var rowBlock = selectedRows?.Rows[rowID];
+      if (rowBlock != null)
+      {
+        BocList.UnselectRow (bocList, rowBlock, isRowHighlightingEnabled);
+      }
+    }
+    
+    //  Start over with a new array
+    selectedRows?.Clear();
+  }
+
+  //  Unselects a row.
+  //  Removes the row frin the _bocList_selectedRows array and decrements _bocList_selectedRowsLength.
+  //  bocList: The BocList to which the row belongs.
+  //  rowBlock: The row to be unselected.
+  //  isRowHighlightingEnabled: true to update the row's css-class.
+  public static UnselectRow(bocList: HTMLElement, rowBlock: BocList_RowBlock, isRowHighlightingEnabled: boolean): void
+  {
+    //  Remove currentRow from list
+    var selectedRows = BocList._selectedRows[bocList.id];
+    if (selectedRows != undefined)
+    {
+      selectedRows.Rows[rowBlock.SelectorControl.id] = null;
+      selectedRows.Length--;
+    }
+
+    // Unselect currentRow
+    rowBlock.SelectorControl.checked = false;
+    if (isRowHighlightingEnabled)
+      $(rowBlock.Row).removeClass(BocList.TrClassNameSelected);
+
+    if (selectedRows != undefined)
+    {
+      BocList.ClearSelectAllRowsSelector (selectedRows);
+    }
+  }
+
+  public static SetSelectAllRowsSelectorOnDemand(selectedRows: BocList_SelectedRows): void
+  {
+    if (selectedRows.DataRowCount == selectedRows.Length && selectedRows.DataRowCount > 0)
+      selectedRows.SelectAllSelectorControls?.each (function (this: JQuery) { this.checked = true; });
+  }
+
+  public static ClearSelectAllRowsSelector(selectedRows: BocList_SelectedRows)
+  {
+    selectedRows?.SelectAllSelectorControls?.each (function (this: JQuery) { this.checked = false; });
+  }
+
+  //  Event handler for the selection selectorControl in the title row.
+  //  Applies the checked state of the title's selectorControl to all data rows' selectu=ion selectorControles.
+  //  bocList: The BocList to which the selectorControl belongs.
+  //  selectRowControlName: The name of the row selector controls.
+  //  isRowHighlightingEnabled: true to enable highting of the rows
+  public static OnSelectAllSelectorControlClick(bocList: HTMLElement, selectAllSelectorControl: JQuery, isRowHighlightingEnabled: boolean): void
+  {
+    var selectedRows = BocList._selectedRows[bocList.id];
+
+    if (selectedRows?.Selection != BocList.rowSelectionMultiple)
+      return;
+    //  BocList_SelectRow will increment the length, therefor initialize it to zero.
+    if (selectAllSelectorControl.checked)
+      selectedRows.Length = 0;
+  
+    selectedRows.SelectRowSelectorControls?.each (function (this: JQuery)
```

Comment:
We might be missing a null check here
```suggestion
    // TODO: Missing null check?
    selectedRows.SelectRowSelectorControls!.each (function (this: JQuery)
```